src/vaults/BalanceSheet.sol

```diff
@@ -124,7 +126,29 @@ contract BalanceSheet is Auth, Recoverable, IBalanceSheet, IBalanceSheetGatewayH
         _issue(poolId, scId, to, pricePoolPerShare, shares);
     }
 
+    /// @inheritdoc IBalanceSheet
+    function transferSharesFrom(PoolId poolId, ShareClassId scId, address from, address to, uint256 amount)
```

Comment:
Thinking about this now, should we remove `triggerRequestRedeem` from `AsyncRequests`, and just in the future allow a custom manager to be developed that integrates the `BalanceSheet` and calls `BalanceSheet.transferSharesFrom` + `vault.requestRedeem` with `controller=user`?